src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -420,6 +551,55 @@ private ODataNestedResourceInfo ReadNestedResourceInfo()
             return link;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeletedResource"/>
+        /// </summary>
+        /// <returns>An <see cref="MaterializerDeletedEntry"/> of the read <see cref="ODataDeletedResource"/>.</returns>
+        private MaterializerDeletedEntry ReadDeletedResource()
+        {
+            this.ExpectState(ODataReaderState.DeletedResourceStart);
+
+            ODataDeletedResource result = (ODataDeletedResource)this.reader.Item;
+            MaterializerDeletedEntry entry;
+
+            if (result != null)
+            {
+                entry = MaterializerDeletedEntry.CreateDeletedEntry(
+                    result,
+                    this.materializerContext);
+
+                if (this.items?.Count > 0)

```

Comment:
See my comment here https://github.com/OData/odata.net/pull/2627/files#r1169891283

Reply:
the stack is only used when we are reading a delta payload. If we are just reading a single payload then this.items will be null hence the check.